src/entrypoints/recorder.ts

```diff
@@ -278,53 +278,58 @@ function initXhrObserver(cb: networkCallback, win: IWindow, options: Required<Ne
+                        const headers = rawHeaders.trim().split(/[\r\n]+/)
+                        headers.forEach((line) => {
+                            const parts = line.split(': ')
+                            const header = parts.shift()
+                            const value = parts.join(': ')
+                            if (header) {
+                                responseHeaders[header] = value
+                            }
                         })
-                    ) {
-                        networkRequest.responseBody = _tryReadXHRBody({ body: xhr.response, options, url })
-                    }
-                    getRequestPerformanceEntry(win, 'xmlhttprequest', req.url, start, end)
-                        .then((entry) => {
-                            const requests = prepareRequest({
-                                entry,
-                                method: method,
-                                status: xhr?.status,
-                                networkRequest,
-                                start,
-                                end,
-                                url: url.toString(),
-                                initiatorType: 'xmlhttprequest',
+                        if (recordResponseHeaders) {
+                            networkRequest.responseHeaders = responseHeaders
+                        }
+                        if (
+                            shouldRecordBody({
+                                type: 'response',
+                                headers: responseHeaders,
+                                url,
+                                recordBody: options.recordBody,
                             })
-                            cb({ requests })
-                        })
-                        .catch(() => {
-                            //
-                        })
-                })
+                        ) {
+                            networkRequest.responseBody = _tryReadXHRBody({ body: xhr.response, options, url })
+                        }
+                        getRequestPerformanceEntry(win, 'xmlhttprequest', req.url, start, end)
+                            .then((entry) => {
+                                const requests = prepareRequest({
+                                    entry,
+                                    method: method,
+                                    status: xhr?.status,
+                                    networkRequest,
+                                    start,
+                                    end,
+                                    url: url.toString(),
+                                    initiatorType: 'xmlhttprequest',
+                                })
+                                cb({ requests })
+                            })
+                            .catch(() => {
+                                //
+                            })
+                    },
+                    { passive: true }
```

Comment:
can i be precious and we not change this... changes in here can really break customers' sites so i change it very carefully 🙈

Reply:
Ah good catch